var/spack/repos/builtin/packages/precice/package.py

```diff
@@ -33,38 +33,68 @@ class Precice(CMakePackage):
     variant('python', default=False, description='Enable Python support')
     variant('shared', default=True, description='Build shared libraries')
 
-    # Not yet
-#    variant(
-#        'float', default=False,
-#        description='Use single precision for field data exchange')
-#    variant(
-#        'int64',
-#        default=False, description='Use 64-bit integers for indices')
-
     depends_on('cmake@3.5:', type='build')
-    depends_on('cmake@3.9.6:', type='build', when='@1.4:')
+    depends_on('cmake@3.10.2:', type='build', when='@1.4:')
     depends_on('boost@1.60.0:')
     depends_on('boost@1.65.1:', when='@1.4:')
     depends_on('eigen@3.2:')
+    depends_on('eigen@:3.3.7', type='build', when='@:1.5')  # bug in prettyprint
     depends_on('libxml2')
     depends_on('mpi', when='+mpi')
     depends_on('petsc@3.6:', when='+petsc')
-    depends_on('python@2.7', when='+python', type=('build', 'run'))
+    depends_on('python@2.7.16', when='+python', type=('build', 'run'))
```

Comment:
It doesn't work with any other version of Python? Python 2 will no longer be supported by the end of the year.

Reply:
Currently we only test python 2, however, we plan to upgrade to python 3 soon for obvious reasons.
Another reason for fixing the python version is to simplify locating the numpy headers here:

https://github.com/spack/spack/pull/12356/files#diff-8b6a38871241552be40b51937352aad7R89-R90

Reply:
If any version of Python 2 works, you want:
```suggestion
    depends_on('python@:2', when='+python', type=('build', 'run'))
```
If only Python 2.7 works, you want:
```suggestion
    depends_on('python@2.7:2.8', when='+python', type=('build', 'run'))
```